### DISCUSSION TITLE
Enabling more flags to check for concurrency compatibility

### DISCUSSION THREAD

**Discussion by mattmassicotte** [SwiftPackageIndex/SwiftPackageIndex-Server#3442] [Ideas]
Hello friends! There are two additional flags that, while technically can be source-incompatible, can also have a meaningfully-positive impact on warnings produced by strict concurrency. 

`InferSendableFromCaptures`: https://github.com/swiftlang/swift-evolution/blob/main/proposals/0418-inferring-sendable-for-methods.md
`GlobalActorIsolatedTypesUsability`: https://github.com/swiftlang/swift-evolution/blob/main/proposals/0434-global-actor-isolated-types-usability.md

I've had to use both of these to address warnings that I would otherwise not be able to fix. I think it may make sense to consider enabling these as well. Thoughts?

**Comment by finestructure** [comment]
Hey Matt, I saw your post yesterday and bookmarked it because I thought we should turn these on :)

**Comment by mattmassicotte** [comment]
Yeah I think these could potentially make a lot of sense. I've frequently encounterd situations where `GlobalActorIsolatedTypesUsability` will help, and I would be really surprised if any package hit the source-incompatibility problems with it.

I'm not as familiar with `InferSendableFromCaptures`, but I say go for it, how bad could it be 😅

Now, seperately, I just finished an initial migration of a package from Swift 5 mode + this stuff to Swift 6, and changing this mode actually uncovered a number of additional issues. I think it is conceivable that it is a bug that warnings would only surface in Swift 6 mode, but it's hard to be 100% sure. I'm going to file that one and I'll follow up.
